mfr/extensions/tabular/libs/h5py_scipy_tools.py

```diff
@@ -0,0 +1,75 @@
+from collections import OrderedDict
+
+import h5py
+import scipy.io
+
+from mfr.extensions.tabular import exceptions
+from mfr.extensions.tabular.utilities import data_population, header_population
+
+def build_sheets(name, list_data, sheets):
+    # add default header to fix rendering of some data formats
+    header = [i for i in range(1, 1 + len(list_data[0]))]
+    values = data_population(list_data, headers=header)
+    header = header_population(header)
+    sheets[str(name)] = (header, values)
+    return sheets
+
+def mat_v73(fp):
+    """Read and convert a mat v7.3+ file to JSON format using h5py
+    :param fp: File pointer object
+    :return: tuple of table headers and data
+    """
+
+    workbook = h5py.File(fp.name, 'r')
+
+    if not workbook:
+        raise(exceptions.EmptyTableError("Cannot render an empty mat file"))
```

Comment:
Similarly here, `h5py.File()` [doc](http://docs.h5py.org/en/latest/high/file.html#File) is a constructor so it will either raise an exception or return an object which can never be `None`. We can use something like "Warning: the .mat file is empty" as mentioned below.